File: dotnet-mcp/McpServer/Models/McpModels.cs
using System.Text.Json.Serialization;

namespace McpServer.Models;

// Base class for all MCP requests
public abstract class McpRequest
{
    [JsonPropertyName("id")]
    public string Id { get; set; } = Guid.NewGuid().ToString();
    
    [JsonPropertyName("version")]
    public string Version { get; set; } = "0.1";
    
    [JsonPropertyName("type")]
    public abstract string Type { get; }
}

// Base class for all MCP responses
public abstract class McpResponse
{
    [JsonPropertyName("id")]
    public string Id { get; set; } = string.Empty;
    
    [JsonPropertyName("type")]
    public abstract string Type { get; }
}

// Session creation request
public class SessionCreateRequest : McpRequest
{
    [JsonPropertyName("type")]
    public override string Type => "session-create-request";
    
    [JsonPropertyName("attributes")]
    public SessionCreateAttributes Attributes { get; set; } = new();
}

public class SessionCreateAttributes
{
    [JsonPropertyName("access_token")]
    public string? AccessToken { get; set; }
}

// Session creation response
public class SessionCreateResponse : McpResponse
{
    [JsonPropertyName("type")]
    public override string Type => "session-create-response";
    
    [JsonPropertyName("session_id")]
    public string SessionId { get; set; } = Guid.NewGuid().ToString();
}

// Text Completion request
public class TextCompletionRequest : McpRequest
{
    [JsonPropertyName("type")]
    public override string Type => "text-completion-request";
    
    [JsonPropertyName("session_id")]
    public string SessionId { get; set; } = string.Empty;
    
    [JsonPropertyName("inputs")]
    public TextCompletionInputs Inputs { get; set; } = new();
}

public class TextCompletionInputs
{
    [JsonPropertyName("prompt")]
    public string Prompt { get; set; } = string.Empty;
    
    [JsonPropertyName("temperature")]
    public double? Temperature { get; set; } = 0.7;
    
    [JsonPropertyName("max_tokens")]
    public int? MaxTokens { get; set; } = 1000;
}

// Text Completion response
public class TextCompletionStreamResponse : McpResponse
{
    [JsonPropertyName("type")]
    public override string Type => "text-completion-stream-response";
    
    [JsonPropertyName("delta")]
    public TextCompletionDelta Delta { get; set; } = new();
    
    [JsonPropertyName("usage")]
    public TextCompletionUsage? Usage { get; set; }
}

public class TextCompletionDelta
{
    [JsonPropertyName("text")]
    public string Text { get; set; } = string.Empty;
    
    [JsonPropertyName("finish_reason")]
    public string? FinishReason { get; set; }
}

public class TextCompletionUsage
{
    [JsonPropertyName("prompt_tokens")]
    public int PromptTokens { get; set; }
    
    [JsonPropertyName("completion_tokens")]
    public int CompletionTokens { get; set; }
    
    [JsonPropertyName("total_tokens")]
    public int TotalTokens { get; set; }
}

// Session close request
public class SessionCloseRequest : McpRequest
{
    [JsonPropertyName("type")]
    public override string Type => "session-close-request";
    
    [JsonPropertyName("session_id")]
    public string SessionId { get; set; } = string.Empty;
}

// Session close response
public class SessionCloseResponse : McpResponse
{
    [JsonPropertyName("type")]
    public override string Type => "session-close-response";
}

// Error response
public class ErrorResponse : McpResponse
{
    [JsonPropertyName("type")]
    public override string Type => "error";
    
    [JsonPropertyName("error")]
    public McpError Error { get; set; } = new();
}

public class McpError
{
    [JsonPropertyName("message")]
    public string Message { get; set; } = string.Empty;
    
    [JsonPropertyName("code")]
    public string Code { get; set; } = string.Empty;
}

File: docker-mcp/src/models/McpModels.js
/**
 * MCP Models - Defines request and response models for the Model Context Protocol
 */
const { v4: uuidv4 } = require('uuid');

/**
 * Base class for all MCP requests
 */
class McpRequest {
  constructor(type) {
    this.id = uuidv4();
    this.version = '0.1';
    this.type = type;
  }
}

/**
 * Base class for all MCP responses
 */
class McpResponse {
  constructor(requestId, type) {
    this.id = requestId;
    this.type = type;
  }
}

/**
 * Session creation request
 */
class SessionCreateRequest extends McpRequest {
  constructor(attributes = {}) {
    super('session-create-request');
    this.attributes = attributes;
  }
}

/**
 * Session creation attributes
 */
class SessionCreateAttributes {
  constructor(accessToken = null) {
    this.access_token = accessToken;
  }
}

/**
 * Session creation response
 */
class SessionCreateResponse extends McpResponse {
  constructor(requestId, sessionId) {
    super(requestId, 'session-create-response');
    this.session_id = sessionId;
  }
}

/**
 * Text completion request
 */
class TextCompletionRequest extends McpRequest {
  constructor(sessionId, inputs = {}) {
    super('text-completion-request');
    this.session_id = sessionId;
    this.inputs = inputs;
  }
}

/**
 * Text completion inputs
 */
class TextCompletionInputs {
  constructor(prompt = '', temperature = 0.7, maxTokens = 1000) {
    this.prompt = prompt;
    this.temperature = temperature;
    this.max_tokens = maxTokens;
  }
}

/**
 * Text completion stream response
 */
class TextCompletionStreamResponse extends McpResponse {
  constructor(requestId, delta = {}, usage = null) {
    super(requestId, 'text-completion-response');
    this.delta = delta;
    this.usage = usage;
  }
}

/**
 * Text completion delta
 */
class TextCompletionDelta {
  constructor(text = '', finishReason = null) {
    this.text = text;
    this.finish_reason = finishReason;
  }
}

/**
 * Text completion usage
 */
class TextCompletionUsage {
  constructor(promptTokens = 0, completionTokens = 0, totalTokens = 0) {
    this.prompt_tokens = promptTokens;
    this.completion_tokens = completionTokens;
    this.total_tokens = totalTokens;
  }
}

/**
 * Session close request
 */
class SessionCloseRequest extends McpRequest {
  constructor(sessionId) {
    super('session-close-request');
    this.session_id = sessionId;
  }
}

/**
 * Session close response
 */
class SessionCloseResponse extends McpResponse {
  constructor(requestId) {
    super(requestId, 'session-close-response');
  }
}

/**
 * Error response
 */
class ErrorResponse extends McpResponse {
  constructor(requestId, code, message) {
    super(requestId, 'error');
    this.error = {
      code,
      message
    };
  }
}

module.exports = {
  McpRequest,
  McpResponse,
  SessionCreateRequest,
  SessionCreateAttributes,
  SessionCreateResponse,
  TextCompletionRequest,
  TextCompletionInputs,
  TextCompletionStreamResponse,
  TextCompletionDelta,
  TextCompletionUsage,
  SessionCloseRequest,
  SessionCloseResponse,
  ErrorResponse
};